csm_web/scheduler/views.py

```diff
@@ -73,10 +73,10 @@ def get_queryset(self):
     @action(detail=True)
     def sections(self, request, pk=None):
         course = get_object_or_error(self.get_queryset(), pk=pk)
-        if request.query_params.get('grouped'):
-            return Response({day: SectionSerializer(sorted(group, key=lambda section: section.spacetime._start_time), many=True).data for day, group in groupby(
-                course.section_set.all().order_by('spacetime___day_of_week'), lambda section: section.spacetime.get__day_of_week_display())})
-        return Response(SectionSerializer(course.section_set.all(), many=True).data)
+        #TODO: Clean this up
+        return Response(dict(sorted(((day, SectionSerializer(group, many=True).data) for day, group in groupby(
```

Comment:
`dict(sorted(...))` kind of rubs me the wrong way - could you use an `OrderedDict` instead?

Reply:
Had the same instinct at first, but `OrderedDict` only preserves insertion order and provides a mechanism to move a key to the end of the ordering. `OrderedDict` provides no mechanism for sorting keys, but we want the keys to be sorted in weekday order for ease of consumption by the frontend. Since Python 3.7 normal dicts are guaranteed to maintain keys in insertion order, but the only way to _get_ the insertion order we want is by sorting.